lua/entities/base_wire_entity.lua

```diff
@@ -288,29 +309,45 @@ end
 -- It allows us to optionally send values rather than entire strings, which saves networking
 -- It also allows us to only update overlays when someone is looking at the entity.
 
+local function cleanInvalidPlayers()
+	for k, v in ipairs( self.playersRequestingOverlayNumeric ) do
+		if !IsValid(v) then table.remove( self.playersRequestingOverlayNumeric, k ) end
+	end
+end
+
 function ENT:SetOverlayText( txt )
 	if not self.OverlayData then
 		self.OverlayData = {}
 	end
-
 	if txt and #txt > 12000 then
 		txt = string.sub(txt,1,12000) -- I have tested this and 12000 chars is enough to cover the entire screen at 1920x1080. You're unlikely to need more
 	end
-
+	if txt == self.OverlayData.txt then return end
 	self.OverlayData.txt = txt
-
-	if not self.OverlayData_UpdateTime then	self.OverlayData_UpdateTime = {} end
-	self.OverlayData_UpdateTime.time = CurTime()
+	if CLIENT then return end
+	self.lastUpdate = CurTime()
+	if #self.playersRequestingOverlayNumeric == 0 then return end
+	cleanInvalidPlayers()
+	net.Start( "wire_overlay_txt", true )
+		net.WriteEntity( self )
+		net.WriteString( self.OverlayData.txt )
+	net.Send(self.playersRequestingOverlayNumeric)
 end
 
 function ENT:SetOverlayData( data )
-	self.OverlayData = data
-	if self.OverlayData.txt and #self.OverlayData.txt > 12000 then
+	if data and self.OverlayData.txt and #self.OverlayData.txt > 12000 then
 		self.OverlayData.txt = string.sub(self.OverlayData.txt,1,12000)
 	end
-
-	if not self.OverlayData_UpdateTime then	self.OverlayData_UpdateTime = {} end
-	self.OverlayData_UpdateTime.time = CurTime()
+	if data == self.OverlayData then return end
```

Comment:
This check won't work because fields of the table may change, but the same table may be inserted. Either the fields should all be checked or this check should be removed.

Reply:
you would think checking a table against a table would internally hash it or something.

Reply:
If the table's hash changed everytime its fields changed, that'd break everything.

Reply:
That's what __eq is for

Reply:
so should i use https://wiki.facepunch.com/gmod/Global.rawequal instead of `==`

Reply:
No, that just makes a comparison ignore __eq. These tables don't have __eq to begin with. The only solution here is to loop over the fields and compare them or implement __eq for every OverlayData.

Reply:
Or remove the check

Reply:
well you CAN crc32 it, but that sounds expensive.

Reply:
the best is probably just to remove the comparison, looping over it all is probably not worth the minor saving of a netmessage or 2.